lib/jetstream/pull_consumer.ex

```diff
@@ -87,26 +88,24 @@ defmodule Jetstream.PullConsumer do
       See `Supervisor`.
       """
       @spec child_spec(
-              init_arg :: %{
-                :connection_name => pid() | atom(),
-                :stream_name => binary(),
-                :consumer_name => binary(),
-                optional(:connection_retry_timeout) => pos_integer(),
-                optional(:connection_retires) => pos_integer()
-              }
+              init_arg ::
+                [settings: Jetstream.PullConsumer.settings()]
+                | [
+                    settings: Jetstream.PullConsumer.settings(),
+                    options: GenServer.options()
+                  ]
```

Comment:
I do not understand the motivation behind this change. Take a look at how Elixir passes GenServer options and how child_spec definition looks like:
- https://hexdocs.pm/elixir/1.13/Supervisor.html
- https://github.com/elixir-lang/elixir/blob/89e57882e120e77d755477b96ecfa15032dd582d/lib/elixir/lib/gen_server.ex#L754-L761
- https://github.com/elixir-ecto/connection/pull/18/files

Reply:
I thought it would be nice to be able to reference the PullConsumer by its name for convenience (now I think it would be better just to simply add the `:name` key to the `settings`). 

But on second thought, after looking at some examples, I agree with you, I think we don't want this in `child_spec`